frontend/src/components/FifthExamAdminTable/FifthExamAdminTable.tsx

```diff
@@ -357,27 +290,94 @@ const EnhancedTableToolbar = (props: EnhancedTableToolbarProps) => {
 };
 
 function FifthExamAdminTable(props) {
-  const [open, setOpen] = React.useState(false);
+  const dispatch = useDispatch();
+
+  const [showTopicDialog, setShowTopicDialog] = React.useState(false);
   const [currentAnnotation, setCurrentAnnotation] = React.useState('');
+  const [showDeclineReasonDialog, setShowDeclineReasonDialog] = React.useState(false);
+  const [currentDeclineReason, setCurrentDeclineReason] = React.useState('');
+  const [currentSelectedExamId, setCurrentSelectedExamId] = React.useState('');
 
   const [order, setOrder] = React.useState<Order>('asc');
-  const [orderBy, setOrderBy] = React.useState<keyof Data>('studentName');
-  const [selected, setSelected] = React.useState<readonly string[]>([]);
+  const [orderBy, setOrderBy] = React.useState<keyof ExamInterface>('studentName');
+  const [selected, setSelected] = React.useState<readonly any[]>([]);
   const [page, setPage] = React.useState(0);
   const [dense, setDense] = React.useState(false);
   const [rowsPerPage, setRowsPerPage] = React.useState(5);
 
+  const rows: ExamInterface[] = useSelector((state: RootState) => state.fithExamAdminReducer.allExams )
+
   const showFullTopic = (annotation) => {
-    setOpen(true);
+    setShowTopicDialog(true);
     setCurrentAnnotation(annotation);
   };
   const hideFullTopic = () => {
-    setOpen(false);
+    setShowTopicDialog(false);
+  };
+  const hideDeclineReasonAndSendApiRequest = () => {
+    setShowDeclineReasonDialog(false);
+
+    let body = {examId: currentSelectedExamId, approved: false, reason: currentDeclineReason}
+    sendApiRequest('/api/abitur/setApprovalState', 'POST', body)
+      .then((response) => {
+        if (response.status != 200) {
+          console.log('Fehler beim Ablehnen: Http-Statuscode: ' + response.status)
+        }
+      });
+    //no need for page reloading
+    rows.map((row) => {
+      if(String(row.examId) == currentSelectedExamId){
+        row.approved=false;
+      }
+    });
+  };
+  const handleCurrentDeclineReasonOnChange= (event) => {
+    setCurrentDeclineReason(event.target.value);
   };
 
+  const history = useHistory();
+  handleEditRowClick = (selectedRows) => {
+    const selectedRowData = rows.find((row) => {
+      // @ts-ignore
+      if(row.examId === selectedRows[0]){
+        return true;
+      }
+    })
+    setPreFilledDataIn5PKFormEditedByAdmin(selectedRowData)
+    history.push('/admin/pruefungskomponente/editStudentApplication')
+  }
+
+  handleOnClickApprove= (event: React.MouseEvent<unknown>, isApproved: boolean) => {
```

Comment:
also kann man jetzt doch nicht mehr als einen Antrag gleichzeitig genehmigen?

Reply:
doch er geht in der for-Schleife alle durch, die ausgewählt sind (selected heißt die Variable)